src/app/entity/entity-mapper.service.ts

```diff
@@ -48,6 +48,29 @@ export class EntityMapperService {
     );
   }
 
+  /**
+   * Loads a list of Entity from the database whose IDs contain the prefix of the given resultEntity class.
+   *
+   * @param entityType the entity's class type (this would just be <code>Entity</code> for a pure entity).
+   * @returns A Promise containing an array with the entities.
+   */
+  public loadAll<T extends Entity>(entityType: { new(id: string): T; }): Promise<T[]> {
+    let newEntity = new entityType('');
+    return this._db.getAll(newEntity.getPrefix()).then(
```

Comment:
Can we somehow get the prefix from the class without creating an instance?  (although I did similarly ugly stuff in `load` - which I would like to get rid of)

Confusing to create a dummy instance here and again in line 64, rather than right before line 63.